src/bsp/src/stm32/src/interloc/src/InterlocStateHandler.cpp

```diff
@@ -2,9 +2,22 @@
 #include <bsp/BSPContainer.h>
 
 InterlocStateHandler::InterlocStateHandler() :
-    m_currentState(&InterlocStateContainer::getState(InterlocStates::DEFAULT)) {}
+    m_currentStateName(InterlocStates::DEFAULT),
+    m_currentState(&InterlocStateContainer::getState(InterlocStates::DEFAULT)),
+    m_stateTracer(m_stateTracerData.data(), MAX_TRACER_TRANSITIONS) {}
 
-void InterlocStateHandler::setState(InterlocStates state) {
+void InterlocStateHandler::setState(InterlocStates state, InterlocEvent event) {
+    if (m_stateTracer.isFull()) {
+        m_stateTracer.pop();
+    }
```

Comment:
Why the need to pop if m_stateTracer is a circular queue?

Reply:
The queue won't push if it's full. So we clear the oldest data before pushing